File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/pullrequestlist/columns/TitleColumnInfo.java
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow.pullrequestlist.columns;

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity;
import com.intellij.util.ui.ColumnInfo;
import org.jetbrains.annotations.Nullable;

public class TitleColumnInfo extends ColumnInfo<PullRequestEntity, String> {
    public TitleColumnInfo() {
        super("title");
    }

    @Nullable
    @Override
    public String valueOf(PullRequestEntity pullRequest) {
        return pullRequest.getTitle();
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/presentation/pullrequestlist/columns/TitleColumnInfo.kt
package com.duxtinto.intellij.plugin.github.codereviews.presentation.pullrequestlist.columns

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity
import com.intellij.util.ui.ColumnInfo

class TitleColumnInfo : ColumnInfo<PullRequestEntity, String>(ColumnInfoFactory.ColumnNames.TITLE.value) {

    override fun valueOf(pullRequest: PullRequestEntity): String? {
        return pullRequest.title
    }
}
